AmplifyPlugins/Core/AWSPluginsCore/Model/Support/QueryPredicate+GraphQL.swift

```diff
@@ -98,6 +100,15 @@ extension QueryPredicate {
     }
 }
 
+extension QueryPredicateConstant: GraphQLFilterConvertible {
+    func graphQLFilter(for modelSchema: ModelSchema?) -> GraphQLFilter {
+        if self == .all {
+            return [:]
```

Comment:
just thinking whether the return value should just be `nil` and the method return `GraphQLFilter?` type. I think this is keeping it `GraphQLFilter` is fine since FilterDecorator should be checking if the filter is empty anyways, just to be safe.

Reply:
Hmm I think it would be useful to change if we had a use case of detecting nil vs an empty GraphQLFilter.  In this case, of QueryPredicateConstant.all, i think that returning [:] is an accurate representation of `.all` -- that is, there is no filter -- so at this point, i'm not seeing a strong case for changing this to an optional.